agent-bootstrap/src/main/java/io/opentelemetry/auto/bootstrap/instrumentation/decorator/HttpServerTracer.java

```diff
@@ -195,6 +201,44 @@ protected void onRequest(final Span span, final REQUEST request) {
     // TODO set resource name from URL.
   }
 
+  protected void onConnectionAndRequest(Span span, CONNECTION connection, REQUEST request) {
+    String flavor = flavor(connection, request);
+    if (flavor != null) {
+      SemanticAttributes.HTTP_FLAVOR.set(span, flavor);
+    }
+    SemanticAttributes.HTTP_CLIENT_IP.set(span, clientIP(connection, request));
+  }
+
+  protected String clientIP(CONNECTION connection, REQUEST request) {
+    // try Forwarded
+    String forwarded = requestHeader(request, "Forwarded");
+    if (forwarded != null) {
+      // may be split by ; or ,
+      for (String forwardedValues : forwarded.split(";|(, *)")) {
+        String[] split = forwardedValues.split("=", 2);
+        if (split.length != 2) {
+          continue;
+        }
+        if ("for".equalsIgnoreCase(split[0]) && !split[1].isEmpty()) {
+          return split[1];
+        }
+      }
+    }
+
+    // try X-Forwarded-For
+    forwarded = requestHeader(request, "X-Forwarded-For");
+    if (forwarded != null) {
+      // may be split by ,
+      String[] split = forwarded.split(", *", 2); // only need the first
```

Comment:
should be slightly faster to split by `,` and then trim result

Reply:
I got rid of the regex in the x-forwarded-for case